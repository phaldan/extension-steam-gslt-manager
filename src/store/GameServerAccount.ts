import { action, observable } from 'mobx';

export default class GameServerAccount {
  public steamid: string;
  public appid: number;

  @observable
  public expired: boolean = false;

  @observable
  public lastLogon?: Date;

  @observable
  public memo: string;

  @observable
  public token: string;

  @observable
  public locked: boolean = false;

  constructor({ steamid, appid, token, memo, lastLogon, expired }) {
    this.steamid = steamid;
    this.appid = appid;
    this.token = token;
    this.memo = memo;
    this.lastLogon = lastLogon ? new Date(lastLogon) : undefined;
    this.expired = expired;
  }

  @action
  public toggleLock(locked: boolean) {
    this.locked = locked;
  }

  @action
  public updateFromJson({ appid, token, memo, lastLogon, expired }) {
    this.appid = appid;
    this.token = token;
    this.memo = memo;
    this.lastLogon = lastLogon;
    this.expired = expired;
  }
}
