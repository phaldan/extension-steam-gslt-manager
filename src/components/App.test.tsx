import { shallow } from 'enzyme';
import { action, computed, observable } from 'mobx';
import * as React from 'react';
import GameServerAccount from 'store/GameServerAccount';
import GsltStore from 'store/GsltStore';
import delay from 'utils/delay';
import App from './App';

function createGetAllJson() {
  return {
    sessionId: '3D6M733LPVJ1',
    tokens: [
      {
        appid: 730,
        expired: false,
        lastLogon: undefined,
        memo: 'CSGO',
        steamid: '212V16ECZ4HE',
        token: '7FJS3VY2273L',
      }
    ]
  };
}

function createAccount() {
  return new GameServerAccount({
    appid: 730,
    expired: false,
    lastLogon: undefined,
    memo: 'CSGO',
    steamid: '212V16ECZ4HE',
    token: '7FJS3VY2273L',
  });
}

class GsltStoreStub implements GsltStore {

  @observable
  private accounts: GameServerAccount[] = [];

  @observable
  private initialized: boolean = false;

  loadAccounts(): Promise<void> {
    throw new Error("Method not implemented.");
  }
  @computed
  get tokenAccounts(): GameServerAccount[] {
    return this.accounts;
  }
  removeAccount(account: GameServerAccount): Promise<GameServerAccount> {
    throw new Error("Method not implemented.");
  }
  removeAccounts(accounts: GameServerAccount[]): Promise<GameServerAccount>[] {
    throw new Error("Method not implemented.");
  }
  regenerateToken(account: GameServerAccount): Promise<GameServerAccount> {
    throw new Error("Method not implemented.");
  }
  regenerateTokens(accounts: GameServerAccount[]): Promise<GameServerAccount>[] {
    throw new Error("Method not implemented.");
  }
  updateMemo(account: GameServerAccount, memo: string): Promise<GameServerAccount> {
    throw new Error("Method not implemented.");
  }
  createAccounts(amount: number, appid: string, memo: string): Promise<void>[] {
    throw new Error("Method not implemented.");
  }
  @computed
  get isLoggedIn(): boolean {
    return true;
  }
  @computed
  get isInitialized(): boolean {
    return this.initialized;
  }

  @action
  setAccounts(accounts: GameServerAccount[]): void {
    this.accounts = accounts;
  }
  @action
  setInitialized(initialized: boolean): void {
    this.initialized = initialized;
  }
}

describe('<App>', () => {
  let state: GsltStoreStub;
  let target;

  beforeEach(() => {
    state = new GsltStoreStub();
    target = shallow(<App store={state} searchDelay={0} />);
  });

  test('empty no accounts', () => {
    expect(target).toMatchSnapshot();
  });

  test('with empty list of accounts', async () => {
    state.setAccounts([]);
    state.setInitialized(true);
    expect(target.update()).toMatchSnapshot();
  });

  test('success filter by memo', async () => {
    state.setAccounts([createAccount()]);
    state.setInitialized(true);
    target.find('.js-filter').simulate('change', { target: { value: 'CSGO' } });
    await delay(0); // Need because of debounce for search delay
    expect(target.update()).toMatchSnapshot();
  });

  test('success filter by appid', async () => {
    state.setAccounts([createAccount()]);
    state.setInitialized(true);
    target.find('.js-filter').simulate('change', { target: { value: '730' } });
    await delay(0); // Need because of debounce for search delay
    expect(target.update()).toMatchSnapshot();
  });

  test('fail to filter', async () => {
    state.setAccounts([createAccount()]);
    state.setInitialized(true);
    target.find('.js-filter').simulate('change', { target: { value: '1337' } });
    await delay(0); // Need because of debounce for search delay
    expect(target.update()).toMatchSnapshot();
  });

  test('success multiple filter', async () => {
    state.setAccounts([createAccount()]);
    state.setInitialized(true);
    target.find('.js-filter').simulate('change', { target: { value: 'CSGO' } });
    target.find('.js-filter').simulate('change', { target: { value: 'CSGO' } });
    await delay(0); // Need because of debounce for search delay
    expect(target.update()).toMatchSnapshot();
  });

  test('clear search-term', async () => {
    state.setAccounts([createAccount()]);
    state.setInitialized(true);
    target.find('.js-filter').simulate('change', { target: { value: '1337' } });
    await delay(0); // Need because of debounce for search delay
    target.find('.js-clear').simulate('click');
    expect(target).toMatchSnapshot();
  });
});
