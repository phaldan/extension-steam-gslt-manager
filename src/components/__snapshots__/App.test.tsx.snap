// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> clear search-term 1`] = `
<div
  className="container"
>
  <Notification
    actions={
      ActionQueueState {
        "queues": Array [],
        "removeDelay": 3000,
      }
    }
  />
  <Login
    show={false}
  />
  <div
    className="page-header"
  >
    <FormGroup
      bsClass="form-group"
      className="pull-right"
    >
      <Create
        actions={
          ActionQueueState {
            "queues": Array [],
            "removeDelay": 3000,
          }
        }
        store={
          GsltStoreStub {
            "accounts": Array [
              GameServerAccount {
                "appid": 730,
                "expired": false,
                "lastLogon": undefined,
                "locked": false,
                "memo": "CSGO",
                "steamid": "212V16ECZ4HE",
                "token": "7FJS3VY2273L",
              },
            ],
            "initialized": true,
          }
        }
        text="New"
      />
      <Readme />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        href="https://github.com/phaldan/steam-gslt-manager"
      >
        <i
          className="fa fa-github"
        />
         Github
      </Button>
    </FormGroup>
    <h1>
      Steam GSLT manager
    </h1>
  </div>
  <Uncontrolled(TabContainer)
    defaultActiveKey="activeGslt"
    id="gslt-tabs"
  >
    <div>
      <div
        className="form-inline pull-right"
      >
        <FormGroup
          bsClass="form-group"
          className="pull-right"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Filter:
          </ControlLabel>
          <InputGroup
            bsClass="input-group"
          >
            <FormControl
              bsClass="form-control"
              className="js-filter"
              componentClass="input"
              onChange={[Function]}
              placeholder="Memo, Game"
              type="text"
              value=""
            />
            <InputGroupButton
              bsClass="input-group-btn"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="js-clear"
                disabled={false}
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="close"
                />
              </Button>
            </InputGroupButton>
          </InputGroup>
        </FormGroup>
      </div>
      <Nav
        bsClass="nav"
        bsStyle="tabs"
        justified={false}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey="activeGslt"
        >
          Active 
          <span
            className="badge"
          >
            1
          </span>
        </NavItem>
        <NavItem
          active={false}
          disabled={false}
          eventKey="expiredGslt"
        >
          Expired 
          <span
            className="badge"
          >
            0
          </span>
        </NavItem>
      </Nav>
      <TabContent
        animation={true}
        bsClass="tab"
        componentClass="div"
        mountOnEnter={false}
        unmountOnExit={false}
      >
        <TabPane
          bsClass="tab-pane"
          eventKey="activeGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={false}
            items={
              Array [
                GameServerAccount {
                  "appid": 730,
                  "expired": false,
                  "lastLogon": undefined,
                  "locked": false,
                  "memo": "CSGO",
                  "steamid": "212V16ECZ4HE",
                  "token": "7FJS3VY2273L",
                },
              ]
            }
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
        <TabPane
          bsClass="tab-pane"
          eventKey="expiredGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={true}
            items={Array []}
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
      </TabContent>
    </div>
  </Uncontrolled(TabContainer)>
</div>
`;

exports[`<App> empty no accounts 1`] = `
<div
  className="container"
>
  <Notification
    actions={
      ActionQueueState {
        "queues": Array [],
        "removeDelay": 3000,
      }
    }
  />
  <Login
    show={false}
  />
  <div
    className="page-header"
  >
    <FormGroup
      bsClass="form-group"
      className="pull-right"
    >
      <Create
        actions={
          ActionQueueState {
            "queues": Array [],
            "removeDelay": 3000,
          }
        }
        store={
          GsltStoreStub {
            "accounts": Array [],
            "initialized": false,
          }
        }
        text="New"
      />
      <Readme />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        href="https://github.com/phaldan/steam-gslt-manager"
      >
        <i
          className="fa fa-github"
        />
         Github
      </Button>
    </FormGroup>
    <h1>
      Steam GSLT manager
    </h1>
  </div>
  <Uncontrolled(TabContainer)
    defaultActiveKey="activeGslt"
    id="gslt-tabs"
  >
    <div>
      <div
        className="form-inline pull-right"
      >
        <FormGroup
          bsClass="form-group"
          className="pull-right"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Filter:
          </ControlLabel>
          <InputGroup
            bsClass="input-group"
          >
            <FormControl
              bsClass="form-control"
              className="js-filter"
              componentClass="input"
              onChange={[Function]}
              placeholder="Memo, Game"
              type="text"
              value=""
            />
            <InputGroupButton
              bsClass="input-group-btn"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="js-clear"
                disabled={false}
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="close"
                />
              </Button>
            </InputGroupButton>
          </InputGroup>
        </FormGroup>
      </div>
      <Nav
        bsClass="nav"
        bsStyle="tabs"
        justified={false}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey="activeGslt"
        >
          Active 
        </NavItem>
        <NavItem
          active={false}
          disabled={false}
          eventKey="expiredGslt"
        >
          Expired 
        </NavItem>
      </Nav>
      <TabContent
        animation={true}
        bsClass="tab"
        componentClass="div"
        mountOnEnter={false}
        unmountOnExit={false}
      >
        <TabPane
          bsClass="tab-pane"
          eventKey="activeGslt"
        >
          <div
            className="text-center"
          >
            <span
              className="loader"
            />
          </div>
        </TabPane>
        <TabPane
          bsClass="tab-pane"
          eventKey="expiredGslt"
        >
          <div
            className="text-center"
          >
            <span
              className="loader"
            />
          </div>
        </TabPane>
      </TabContent>
    </div>
  </Uncontrolled(TabContainer)>
</div>
`;

exports[`<App> fail to filter 1`] = `
<div
  className="container"
>
  <Notification
    actions={
      ActionQueueState {
        "queues": Array [],
        "removeDelay": 3000,
      }
    }
  />
  <Login
    show={false}
  />
  <div
    className="page-header"
  >
    <FormGroup
      bsClass="form-group"
      className="pull-right"
    >
      <Create
        actions={
          ActionQueueState {
            "queues": Array [],
            "removeDelay": 3000,
          }
        }
        store={
          GsltStoreStub {
            "accounts": Array [
              GameServerAccount {
                "appid": 730,
                "expired": false,
                "lastLogon": undefined,
                "locked": false,
                "memo": "CSGO",
                "steamid": "212V16ECZ4HE",
                "token": "7FJS3VY2273L",
              },
            ],
            "initialized": true,
          }
        }
        text="New"
      />
      <Readme />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        href="https://github.com/phaldan/steam-gslt-manager"
      >
        <i
          className="fa fa-github"
        />
         Github
      </Button>
    </FormGroup>
    <h1>
      Steam GSLT manager
    </h1>
  </div>
  <Uncontrolled(TabContainer)
    defaultActiveKey="activeGslt"
    id="gslt-tabs"
  >
    <div>
      <div
        className="form-inline pull-right"
      >
        <FormGroup
          bsClass="form-group"
          className="pull-right"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Filter:
          </ControlLabel>
          <InputGroup
            bsClass="input-group"
          >
            <FormControl
              bsClass="form-control"
              className="js-filter"
              componentClass="input"
              onChange={[Function]}
              placeholder="Memo, Game"
              type="text"
              value="1337"
            />
            <InputGroupButton
              bsClass="input-group-btn"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="js-clear"
                disabled={false}
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="close"
                />
              </Button>
            </InputGroupButton>
          </InputGroup>
        </FormGroup>
      </div>
      <Nav
        bsClass="nav"
        bsStyle="tabs"
        justified={false}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey="activeGslt"
        >
          Active 
          <span
            className="badge"
          >
            0
          </span>
        </NavItem>
        <NavItem
          active={false}
          disabled={false}
          eventKey="expiredGslt"
        >
          Expired 
          <span
            className="badge"
          >
            0
          </span>
        </NavItem>
      </Nav>
      <TabContent
        animation={true}
        bsClass="tab"
        componentClass="div"
        mountOnEnter={false}
        unmountOnExit={false}
      >
        <TabPane
          bsClass="tab-pane"
          eventKey="activeGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={false}
            items={Array []}
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
        <TabPane
          bsClass="tab-pane"
          eventKey="expiredGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={true}
            items={Array []}
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
      </TabContent>
    </div>
  </Uncontrolled(TabContainer)>
</div>
`;

exports[`<App> success filter by appid 1`] = `
<div
  className="container"
>
  <Notification
    actions={
      ActionQueueState {
        "queues": Array [],
        "removeDelay": 3000,
      }
    }
  />
  <Login
    show={false}
  />
  <div
    className="page-header"
  >
    <FormGroup
      bsClass="form-group"
      className="pull-right"
    >
      <Create
        actions={
          ActionQueueState {
            "queues": Array [],
            "removeDelay": 3000,
          }
        }
        store={
          GsltStoreStub {
            "accounts": Array [
              GameServerAccount {
                "appid": 730,
                "expired": false,
                "lastLogon": undefined,
                "locked": false,
                "memo": "CSGO",
                "steamid": "212V16ECZ4HE",
                "token": "7FJS3VY2273L",
              },
            ],
            "initialized": true,
          }
        }
        text="New"
      />
      <Readme />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        href="https://github.com/phaldan/steam-gslt-manager"
      >
        <i
          className="fa fa-github"
        />
         Github
      </Button>
    </FormGroup>
    <h1>
      Steam GSLT manager
    </h1>
  </div>
  <Uncontrolled(TabContainer)
    defaultActiveKey="activeGslt"
    id="gslt-tabs"
  >
    <div>
      <div
        className="form-inline pull-right"
      >
        <FormGroup
          bsClass="form-group"
          className="pull-right"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Filter:
          </ControlLabel>
          <InputGroup
            bsClass="input-group"
          >
            <FormControl
              bsClass="form-control"
              className="js-filter"
              componentClass="input"
              onChange={[Function]}
              placeholder="Memo, Game"
              type="text"
              value="730"
            />
            <InputGroupButton
              bsClass="input-group-btn"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="js-clear"
                disabled={false}
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="close"
                />
              </Button>
            </InputGroupButton>
          </InputGroup>
        </FormGroup>
      </div>
      <Nav
        bsClass="nav"
        bsStyle="tabs"
        justified={false}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey="activeGslt"
        >
          Active 
          <span
            className="badge"
          >
            1
          </span>
        </NavItem>
        <NavItem
          active={false}
          disabled={false}
          eventKey="expiredGslt"
        >
          Expired 
          <span
            className="badge"
          >
            0
          </span>
        </NavItem>
      </Nav>
      <TabContent
        animation={true}
        bsClass="tab"
        componentClass="div"
        mountOnEnter={false}
        unmountOnExit={false}
      >
        <TabPane
          bsClass="tab-pane"
          eventKey="activeGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={false}
            items={
              Array [
                GameServerAccount {
                  "appid": 730,
                  "expired": false,
                  "lastLogon": undefined,
                  "locked": false,
                  "memo": "CSGO",
                  "steamid": "212V16ECZ4HE",
                  "token": "7FJS3VY2273L",
                },
              ]
            }
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
        <TabPane
          bsClass="tab-pane"
          eventKey="expiredGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={true}
            items={Array []}
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
      </TabContent>
    </div>
  </Uncontrolled(TabContainer)>
</div>
`;

exports[`<App> success filter by memo 1`] = `
<div
  className="container"
>
  <Notification
    actions={
      ActionQueueState {
        "queues": Array [],
        "removeDelay": 3000,
      }
    }
  />
  <Login
    show={false}
  />
  <div
    className="page-header"
  >
    <FormGroup
      bsClass="form-group"
      className="pull-right"
    >
      <Create
        actions={
          ActionQueueState {
            "queues": Array [],
            "removeDelay": 3000,
          }
        }
        store={
          GsltStoreStub {
            "accounts": Array [
              GameServerAccount {
                "appid": 730,
                "expired": false,
                "lastLogon": undefined,
                "locked": false,
                "memo": "CSGO",
                "steamid": "212V16ECZ4HE",
                "token": "7FJS3VY2273L",
              },
            ],
            "initialized": true,
          }
        }
        text="New"
      />
      <Readme />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        href="https://github.com/phaldan/steam-gslt-manager"
      >
        <i
          className="fa fa-github"
        />
         Github
      </Button>
    </FormGroup>
    <h1>
      Steam GSLT manager
    </h1>
  </div>
  <Uncontrolled(TabContainer)
    defaultActiveKey="activeGslt"
    id="gslt-tabs"
  >
    <div>
      <div
        className="form-inline pull-right"
      >
        <FormGroup
          bsClass="form-group"
          className="pull-right"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Filter:
          </ControlLabel>
          <InputGroup
            bsClass="input-group"
          >
            <FormControl
              bsClass="form-control"
              className="js-filter"
              componentClass="input"
              onChange={[Function]}
              placeholder="Memo, Game"
              type="text"
              value="CSGO"
            />
            <InputGroupButton
              bsClass="input-group-btn"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="js-clear"
                disabled={false}
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="close"
                />
              </Button>
            </InputGroupButton>
          </InputGroup>
        </FormGroup>
      </div>
      <Nav
        bsClass="nav"
        bsStyle="tabs"
        justified={false}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey="activeGslt"
        >
          Active 
          <span
            className="badge"
          >
            1
          </span>
        </NavItem>
        <NavItem
          active={false}
          disabled={false}
          eventKey="expiredGslt"
        >
          Expired 
          <span
            className="badge"
          >
            0
          </span>
        </NavItem>
      </Nav>
      <TabContent
        animation={true}
        bsClass="tab"
        componentClass="div"
        mountOnEnter={false}
        unmountOnExit={false}
      >
        <TabPane
          bsClass="tab-pane"
          eventKey="activeGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={false}
            items={
              Array [
                GameServerAccount {
                  "appid": 730,
                  "expired": false,
                  "lastLogon": undefined,
                  "locked": false,
                  "memo": "CSGO",
                  "steamid": "212V16ECZ4HE",
                  "token": "7FJS3VY2273L",
                },
              ]
            }
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
        <TabPane
          bsClass="tab-pane"
          eventKey="expiredGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={true}
            items={Array []}
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
      </TabContent>
    </div>
  </Uncontrolled(TabContainer)>
</div>
`;

exports[`<App> success multiple filter 1`] = `
<div
  className="container"
>
  <Notification
    actions={
      ActionQueueState {
        "queues": Array [],
        "removeDelay": 3000,
      }
    }
  />
  <Login
    show={false}
  />
  <div
    className="page-header"
  >
    <FormGroup
      bsClass="form-group"
      className="pull-right"
    >
      <Create
        actions={
          ActionQueueState {
            "queues": Array [],
            "removeDelay": 3000,
          }
        }
        store={
          GsltStoreStub {
            "accounts": Array [
              GameServerAccount {
                "appid": 730,
                "expired": false,
                "lastLogon": undefined,
                "locked": false,
                "memo": "CSGO",
                "steamid": "212V16ECZ4HE",
                "token": "7FJS3VY2273L",
              },
            ],
            "initialized": true,
          }
        }
        text="New"
      />
      <Readme />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        href="https://github.com/phaldan/steam-gslt-manager"
      >
        <i
          className="fa fa-github"
        />
         Github
      </Button>
    </FormGroup>
    <h1>
      Steam GSLT manager
    </h1>
  </div>
  <Uncontrolled(TabContainer)
    defaultActiveKey="activeGslt"
    id="gslt-tabs"
  >
    <div>
      <div
        className="form-inline pull-right"
      >
        <FormGroup
          bsClass="form-group"
          className="pull-right"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Filter:
          </ControlLabel>
          <InputGroup
            bsClass="input-group"
          >
            <FormControl
              bsClass="form-control"
              className="js-filter"
              componentClass="input"
              onChange={[Function]}
              placeholder="Memo, Game"
              type="text"
              value="CSGO"
            />
            <InputGroupButton
              bsClass="input-group-btn"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="js-clear"
                disabled={false}
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="close"
                />
              </Button>
            </InputGroupButton>
          </InputGroup>
        </FormGroup>
      </div>
      <Nav
        bsClass="nav"
        bsStyle="tabs"
        justified={false}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey="activeGslt"
        >
          Active 
          <span
            className="badge"
          >
            1
          </span>
        </NavItem>
        <NavItem
          active={false}
          disabled={false}
          eventKey="expiredGslt"
        >
          Expired 
          <span
            className="badge"
          >
            0
          </span>
        </NavItem>
      </Nav>
      <TabContent
        animation={true}
        bsClass="tab"
        componentClass="div"
        mountOnEnter={false}
        unmountOnExit={false}
      >
        <TabPane
          bsClass="tab-pane"
          eventKey="activeGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={false}
            items={
              Array [
                GameServerAccount {
                  "appid": 730,
                  "expired": false,
                  "lastLogon": undefined,
                  "locked": false,
                  "memo": "CSGO",
                  "steamid": "212V16ECZ4HE",
                  "token": "7FJS3VY2273L",
                },
              ]
            }
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
        <TabPane
          bsClass="tab-pane"
          eventKey="expiredGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={true}
            items={Array []}
            store={
              GsltStoreStub {
                "accounts": Array [
                  GameServerAccount {
                    "appid": 730,
                    "expired": false,
                    "lastLogon": undefined,
                    "locked": false,
                    "memo": "CSGO",
                    "steamid": "212V16ECZ4HE",
                    "token": "7FJS3VY2273L",
                  },
                ],
                "initialized": true,
              }
            }
          />
        </TabPane>
      </TabContent>
    </div>
  </Uncontrolled(TabContainer)>
</div>
`;

exports[`<App> with empty list of accounts 1`] = `
<div
  className="container"
>
  <Notification
    actions={
      ActionQueueState {
        "queues": Array [],
        "removeDelay": 3000,
      }
    }
  />
  <Login
    show={false}
  />
  <div
    className="page-header"
  >
    <FormGroup
      bsClass="form-group"
      className="pull-right"
    >
      <Create
        actions={
          ActionQueueState {
            "queues": Array [],
            "removeDelay": 3000,
          }
        }
        store={
          GsltStoreStub {
            "accounts": Array [],
            "initialized": true,
          }
        }
        text="New"
      />
      <Readme />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
        href="https://github.com/phaldan/steam-gslt-manager"
      >
        <i
          className="fa fa-github"
        />
         Github
      </Button>
    </FormGroup>
    <h1>
      Steam GSLT manager
    </h1>
  </div>
  <Uncontrolled(TabContainer)
    defaultActiveKey="activeGslt"
    id="gslt-tabs"
  >
    <div>
      <div
        className="form-inline pull-right"
      >
        <FormGroup
          bsClass="form-group"
          className="pull-right"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Filter:
          </ControlLabel>
          <InputGroup
            bsClass="input-group"
          >
            <FormControl
              bsClass="form-control"
              className="js-filter"
              componentClass="input"
              onChange={[Function]}
              placeholder="Memo, Game"
              type="text"
              value=""
            />
            <InputGroupButton
              bsClass="input-group-btn"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="js-clear"
                disabled={false}
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="close"
                />
              </Button>
            </InputGroupButton>
          </InputGroup>
        </FormGroup>
      </div>
      <Nav
        bsClass="nav"
        bsStyle="tabs"
        justified={false}
        pullLeft={false}
        pullRight={false}
        stacked={false}
      >
        <NavItem
          active={false}
          disabled={false}
          eventKey="activeGslt"
        >
          Active 
          <span
            className="badge"
          >
            0
          </span>
        </NavItem>
        <NavItem
          active={false}
          disabled={false}
          eventKey="expiredGslt"
        >
          Expired 
          <span
            className="badge"
          >
            0
          </span>
        </NavItem>
      </Nav>
      <TabContent
        animation={true}
        bsClass="tab"
        componentClass="div"
        mountOnEnter={false}
        unmountOnExit={false}
      >
        <TabPane
          bsClass="tab-pane"
          eventKey="activeGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={false}
            items={Array []}
            store={
              GsltStoreStub {
                "accounts": Array [],
                "initialized": true,
              }
            }
          />
        </TabPane>
        <TabPane
          bsClass="tab-pane"
          eventKey="expiredGslt"
        >
          <List
            actions={
              ActionQueueState {
                "queues": Array [],
                "removeDelay": 3000,
              }
            }
            enableRegenerate={true}
            items={Array []}
            store={
              GsltStoreStub {
                "accounts": Array [],
                "initialized": true,
              }
            }
          />
        </TabPane>
      </TabContent>
    </div>
  </Uncontrolled(TabContainer)>
</div>
`;
